Fix test_pnl_service_redis.py: date-dependent assertions drift (2 failures)

## What fails

- `test_daily_rollup_scheduling`: `AssertionError: assert 12 == 16`
- `test_monthly_rollup_scheduling`: `AssertionError: assert 31 == 1`

## Root cause

The assertions hardcode values that depend on "today's date". The `== 16` assertion was correct when the test was written on the 16th of some month; `== 1` was correct on the 1st. The tests pass only on those specific dates — a classic frozen-clock bug.

## How to reproduce
```bash
pytest tests/unit/test_pnl_service_redis.py::TestPnLServiceRedisStreams::test_daily_rollup_scheduling
```

## Fix

Use `freezegun` or `unittest.mock.patch` on `datetime.now()` to pin the current time inside the test, then assert against the expected value for that pinned instant. Do not rely on the system clock.

```python
from unittest.mock import patch
from datetime import datetime

@patch("spreadpilot_core.pnl.service.datetime")
def test_daily_rollup_scheduling(self, mock_dt):
    mock_dt.now.return_value = datetime(2026, 4, 16, 9, 0, 0)
    ...
```


---
*Detected on CI run for PR #178 (https://github.com/AIgen-Solutions-s-r-l/SpreadPilot/actions/runs/24424919660). These failures are also present on `main` (last 3 runs of `CI Pipeline` are `failure`). They block the CI gate for every PR.*

--- a/tests/unit/test_pnl_service_redis.py
+++ b/tests/unit/test_pnl_service_redis.py
@@ -183,7 +183,9 @@ class TestPnLServiceRedisStreams:
                 is_open = pnl_service._is_market_open()
                 assert not is_open
 
-    @freeze_time("2025-06-29 16:30:00")  # Daily rollup time (4:30 PM ET)
+    # freeze_time interprets the literal as UTC. June = EDT (UTC-4), so to
+    # observe 16:30 ET the UTC clock must be 20:30.
+    @freeze_time("2025-06-29 20:30:00")  # 16:30 ET / 20:30 UTC (EDT = UTC-4)
     @pytest.mark.asyncio
     async def test_daily_rollup_scheduling(self, pnl_service):
         """Test daily rollup scheduling at 16:30 ET."""
@@ -363,7 +365,9 @@ class TestPnLServiceRedisStreams:
                     # Expected - should handle gracefully
                     pass
 
-    @freeze_time("2025-01-01 00:10:00")  # Monthly rollup time
+    # freeze_time interprets the literal as UTC. January = EST (UTC-5), so to
+    # observe 2025-01-01 00:10 ET the UTC clock must be 2025-01-01 05:10.
+    @freeze_time("2025-01-01 05:10:00")  # 00:10 ET / 05:10 UTC (EST = UTC-5)
     @pytest.mark.asyncio
     async def test_monthly_rollup_scheduling(self, pnl_service):
         """Test monthly rollup scheduling on 1st at 00:10 ET."""

